### ISSUE TITLE
Crash on startup with "QPaintDevice: Cannot destroy paint device that is being painted"

### ISSUE THREAD

**Issue by mattipulkkinen** [flathub/org.kde.kdenlive#227]
On a fully up-to-date Fedora Workstation 38, the latest Kdenlive flatpak crashes on start. Trying to start it again, it asks me if I want to reset my configuration file, but even answering "yes" doesn't solve the issue. I didn't see anything about QPaintDevice in #161, so I made a new issue for this.

Running from the command line produces this output:

```
QSocketNotifier: Can only be used with threads started with QThread
qt.qpa.qgnomeplatform: Could not find color scheme  ""
 = = = =
GOT Deleted file:  "/home/user/.var/app/org.kde.kdenlive/config/kdenlive-flatpakrc"
 = = = =
GOT Deleted file:  "/app/share/kxmlgui5/kdenlive/kdenliveui.rc"
QPaintDevice: Cannot destroy paint device that is being painted
```

### `flatpak info org.kde.kdenlive`

```
Kdenlive - Video Editor

          ID: org.kde.kdenlive
         Ref: app/org.kde.kdenlive/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 23.04.1
     License: GPL-3.0-only
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 250,5 MB
     Runtime: org.kde.Platform/x86_64/5.15-22.08
         Sdk: org.kde.Sdk/x86_64/5.15-22.08

      Commit: ed2b74960154415e9d37c049cdd4821c7ea24b3585ee1138463b44da5d4113f7
      Parent: a91744250977d5151f3b78e17a8c3f2c894d604e7a3a3e46b8c67b3d40ec93ec
     Subject: Update 7 modules (3ed1c7de)
        Date: 2023-05-19 09:52:55 +0000

```

### System info:

```
System:
  Kernel: 6.3.3-200.fc38.x86_64 arch: x86_64 bits: 64 Desktop: GNOME v: 44.1
    Distro: Fedora release 38 (Thirty Eight)
Machine:
  Type: Desktop Mobo: ASRock model: X570 Taichi serial: <superuser required>
    UEFI: American Megatrends v: P5.01 date: 01/18/2023
Battery:
  ID-1: hidpp_battery_0 charge: 95% condition: N/A
CPU:
  Info: 8-core AMD Ryzen 7 5800X3D [MT MCP] speed (MHz): avg: 1395
    min/max: 550/4550
Graphics:
  Device-1: AMD Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] driver: amdgpu
    v: kernel
  Display: wayland server: X.Org v: 22.1.9 with: Xwayland v: 22.1.9
    compositor: gnome-shell driver: X: loaded: amdgpu
    unloaded: fbdev,modesetting,radeon,vesa dri: radeonsi gpu: amdgpu
    resolution: 2560x1440~240Hz
  API: OpenGL v: 4.6 Mesa 23.0.3 renderer: AMD Radeon RX 6800 XT (navi21
    LLVM 16.0.1 DRM 3.52 6.3.3-200.fc38.x86_64)
Network:
  Device-1: Intel Wi-Fi 6 AX200 driver: iwlwifi
  Device-2: Intel I211 Gigabit Network driver: igb
  Device-3: Sony DualSense wireless controller (PS5) type: USB
    driver: playstation,snd-usb-audio,usbhid
Drives:
  Local Storage: total: 1.82 TiB used: 1.3 TiB (71.3%)
Info:
  Processes: 596 Uptime: 4h 7m Memory: 31.25 GiB used: 7.24 GiB (23.2%)
  Shell: Zsh inxi: 3.3.26

```

**Comment by hellpe** [comment]
I'm having the exact same issue. The AppImage version launches fine.

I've reported a bug upstream: https://bugs.kde.org/show_bug.cgi?id=470045

**Comment by poeticpenguin** [comment]
I'm not a developer, just a user, but can also confirm this issue on my Fedora 38 installation.

I can't pin it down directly to state, which was the last working version, though.

My output while running it slightly differs, though:

> 
> **Qt: Session management error: None of the authentication protocols specified are supported**
>  = = = =
> GOT Deleted file:  "/home/yuri/.var/app/org.kde.kdenlive/config/kdenlive-flatpakrc"
>  = = = =
> GOT Deleted file:  "/app/share/kxmlgui5/kdenlive/kdenliveui.rc"
> QPaintDevice: Cannot destroy paint device that is being painted

General infos:
>          Kennung: org.kde.kdenlive
>              Ref: app/org.kde.kdenlive/x86_64/stable
>      Architektur: x86_64
>            Zweig: stable
>          Version: 23.04.1
>           Lizenz: GPL-3.0-only
>         Ursprung: flathub
>         Sammlung: org.flathub.Stable
>     Installation: system
>      Installiert: 250,5 MB
> Laufzeitumgebung: org.kde.Platform/x86_64/5.15-22.08
>              Sdk: org.kde.Sdk/x86_64/5.15-22.08
> 
>           Commit: 4e9e1dc0076d44e7c187a13e8c8ce7a154ed6311573d9b238cfb761f72217b68
>           Parent: ed2b74960154415e9d37c049cdd4821c7ea24b3585ee1138463b44da5d4113f7
>          Betreff: Update requests-2.30.0-py3-none-any.whl to 2.31.0 (0c17223c)
>            Datum: 2023-05-22 19:36:14 +0000

**Comment by travier** [comment]
I can not reproduce on a fairly up-to-date Fedora Kinoite so not sure where the issue is.

```
● fedora:fedora/38/x86_64/kinoite
                  Version: 38.20230522.0 (2023-05-22T00:46:46Z)
```

**Comment by hellpe** [comment]
If that can be useful:
```
$ rpm-ostree status
State: idle
Deployments:
● fedora:fedora/38/x86_64/silverblue
                  Version: 38.20230524.0 (2023-05-24T00:42:38Z)
               BaseCommit: 945120a038a3681a1fc13c97fa1904362aec43b77a5b629d8eae375b9fe1cebf
             GPGSignature: Valid signature by 6A51BBABBA3D5467B6171221809A8D7CEB10B464
      RemovedBasePackages: libavfilter-free libavformat-free libpostproc-free libswresample-free libavutil-free libswscale-free libavcodec-free 6.0-4.fc38
                           mesa-va-drivers 23.0.3-3.fc38
          LayeredPackages: ffmpeg gstreamer1-plugins-bad-free-extras
                           gstreamer1-plugins-bad-freeworld gstreamer1-plugins-ugly
                           gstreamer1-vaapi langpacks-fr mesa-va-drivers-freeworld
                           rpmfusion-free-release rpmfusion-nonfree-release
```

**Comment by mattipulkkinen** [comment]
This popped up on [Twitter](https://twitter.com/gamingonlinux/status/1661080345528156174) too. Is there more info needed, or something I should check to see whether it could be a cause, or anything else I could do? I'm reluctant to reinstall my whole OS to see if a fresh installation would fix it, and reinstalling the flatpak itself didn't fix the issue.

**Comment by travier** [comment]
It looks like this breaks on non KDE Desktops as you're on GNOME and above it's Silverblue. Not sure why however.

**Comment by krummm** [comment]
Had this issue as well. I had to install the qt5-graphicaleffects to solve this, which is related to why it works on KDE but not GNOME.

**Comment by krummm** [comment]
I have had this issue with kdenlive directly from my package manager, and was solved by installing "qt5-graphicaleffects" on my system, though I think the package is named "qt5-qtgraphicaleffects" on Fedora. I have had the same behavior as @travier here with it working fine on KDE but not elsewhere, perhaps this package is being installed as a dependency of something else on KDE.

**Comment by hellpe** [comment]
> > qt5-qtgraphicaleffects
> 
> Unfortunately, I already have that package installed.

Same here:
```
$ rpm-ostree install qt5-qtgraphicaleffects
error: "qt5-qtgraphicaleffects" is already provided by: qt5-qtgraphicaleffects-5.15.9-1.fc38.x86_64. Use --allow-inactive to explicitly require it.
```

**Comment by poeticpenguin** [comment]
Just in case, i have changed the laptop (now on AMD Renoir 4800H) and with other Gnome Shell Themes.

I remember that some programs wouldn't load due to the NVIDIA driver - Switching to Intel or AMD via PRIME did the trick.

Is there a way for us to compare our systeminfos?

Does anyone here got it work under Gnome (be it Wayland or X11) on F38 Workstation Edition?

**Comment by tcitworld** [comment]
> Does anyone here got it work under Gnome (be it Wayland or X11) on F38 Workstation Edition?

No issues at launch on Fedora Silverblue 38 (GNOME) here:
<details>
<summary>Startup logs</summary>
<pre>
> flatpak run org.kde.kdenlive
QSocketNotifier: Can only be used with threads started with QThread
profilePath from KdenliveSetting::mltPath:  "/app/share/mlt-7/profiles"
meltPath from KdenliveSetting::rendererPath:  "/app/bin/melt"
Empty metadata for  "audiochannels"
Empty metadata for  "audioconvert"
Empty metadata for  "imageconvert"
Empty metadata for  "glsl.manager"
Empty metadata for  "telecide"
plugin not available: "avfilter.acompressor"
plugin not available: "avfilter.aecho"
plugin not available: "avfilter.agate"
plugin not available: "avfilter.atadenoise"
plugin not available: "avfilter.bs2b"
plugin not available: "avfilter.bwdif"
plugin not available: "avfilter.deblock"
plugin not available: "avfilter.dedot"
plugin not available: "avfilter.deflate"
plugin not available: "avfilter.derain"
plugin not available: "avfilter.doubleweave"
plugin not available: "avfilter.field"
plugin not available: "avfilter.framestep"
plugin not available: "avfilter.fspp"
plugin not available: "avfilter.graphmonitor"
plugin not available: "avfilter.hqdn3d"
plugin not available: "avfilter.inflate"
plugin not available: "avfilter.lagfun"
plugin not available: "avfilter.loudnorm"
plugin not available: "avfilter.mcdeint"
plugin not available: "avfilter.random"
plugin not available: "avfilter.removegrain"
plugin not available: "avfilter.separatefields"
plugin not available: "avfilter.shuffleplanes"
plugin not available: "avfilter.sofalizer"
plugin not available: "avfilter.sr"
plugin not available: "avfilter.tmix"
plugin not available: "avfilter.w3fdif"
plugin not available: "avfilter.weave"
plugin not available: "avfilter.yadif"
plugin not available: "frei0r.baltan"
plugin not available: "frei0r.bgsubtract0r"
plugin not available: "frei0r.delay0r"
plugin not available: "frei0r.delaygrab"
plugin not available: "frei0r.facebl0r"
plugin not available: "frei0r.facedetect"
plugin not available: "frei0r.lightgraffiti"
plugin not available: "frei0r.lightgraffiti"
plugin not available: "frei0r.tehRoxx0r"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "movit.unsharp_mask"
plugin not available: "timewarp"
plugin not available: "region"
=== /// CANNOT ACCESS SPEECH DICTIONARIES FOLDER
QQmlEngine::setContextForObject(): Object already has a QQmlContext
QQmlEngine::setContextForObject(): Object already has a QQmlContext
QQmlEngine::setContextForObject(): Object already has a QQmlContext
QQmlEngine::setContextForObject(): Object already has a QQmlContext
qt.qpa.qgnomeplatform: Could not find color scheme  ""
:::: NOT FOUND DOCUMENT GUIDES !!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!
QQmlEngine::setContextForObject(): Object already has a QQmlContext
qrc:/qml/timeline.qml:497: ReferenceError: proxy is not defined
qrc:/qml/timeline.qml:478: ReferenceError: proxy is not defined
==== NO PREVIOUS TIMELINE
==== CONNECT NEW TIMELINE, MODEL: 0
::::::::::: connecting timeline:  QUuid("{7edbfe19-7a56-4d6f-a18a-169964e90859}") , DUR:  0
root context get sub model new function
HHHHHHHHHHHH
INVALID BIN PLAYLIST...
=== OPENING FILE WITH TRACKS:  5
:::: FOUND TRACTOR WITH TRACKS:  2 

___________________________
=== MODEL DATA CHANGED: MUTE DONE TRACK NOT FOUND!!!
:::: FOUND TRACTOR WITH TRACKS:  2 

___________________________
=== MODEL DATA CHANGED: MUTE DONE TRACK NOT FOUND!!!
:::: FOUND TRACTOR WITH TRACKS:  2 

___________________________
=== MODEL DATA CHANGED: MUTE DONE TRACK NOT FOUND!!!
:::: FOUND TRACTOR WITH TRACKS:  2 

___________________________
=== MODEL DATA CHANGED: MUTE DONE TRACK NOT FOUND!!!
=== GOT THUMB FOR:  -1 x -1
===== ACTIVATING DOCUMENT:  QUuid("{7edbfe19-7a56-4d6f-a18a-169964e90859}") 
::::::::::::::::::::::
qml: item not found
qml: item not found
qml: item not found

** (kdenlive:2): WARNING **: 15:46:10.904: atk-bridge: get_device_events_reply: unknown signature
</pre>
</details>

Got the message (but harmless) when closing the app though
<details>
<summary>Closing logs</summary>
<pre>
==== NO PREVIOUS TIMELINE
==== ABORTING NO TIMELINE AVAILABLE
========== READY FOR TASK DISCARD ON:  3
===== REMOVING MASTER PRODUCER; CURRENT COUNT:  1 
:::::::::::::::::::::::::::
QPaintDevice: Cannot destroy paint device that is being painted

</pre>
</details>

```
❯ flatpak info org.kde.kdenlive

Kdenlive - Éditeur vidéo

          ID: org.kde.kdenlive
         Ref: app/org.kde.kdenlive/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 23.04.1
     License: GPL-3.0-only
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 250,5 Mo
     Runtime: org.kde.Platform/x86_64/5.15-22.08
         Sdk: org.kde.Sdk/x86_64/5.15-22.08

      Commit: bae904ae7faaa2a9486d38d3f314344be9ca81ac31ad982d81427c74df7790d4
      Parent: f746e15f53ec0d163296d2505eae5648a1594881362e0f4f457506c5f00287e7
     Subject: Update urllib3-2.0.2-py3-none-any.whl to 2.0.3 (9d0f750d)
        Date: 2023-06-12 18:23:13 +0000
```


```
● fedora:fedora/38/x86_64/silverblue
                  Version: 38.20230612.0 (2023-06-12T00:45:27Z)
```

**Comment by hellpe** [comment]
@tcitworld My own logs look very similar, except they end abruptly when the crash occurs:

```
$ flatpak run org.kde.kdenlive
QSocketNotifier: Can only be used with threads started with QThread
qt.qpa.qgnomeplatform: Could not find color scheme  ""
profilePath from appDir/../share:  "/app/share/mlt-7/profiles"
meltPath from KdenliveSetting::rendererPath:  "/app/bin/melt"
Empty metadata for  "audiochannels"
Empty metadata for  "audioconvert"
Empty metadata for  "imageconvert"
Empty metadata for  "glsl.manager"
Empty metadata for  "telecide"
plugin not available: "avfilter.acompressor"
plugin not available: "avfilter.aecho"
plugin not available: "avfilter.agate"
plugin not available: "avfilter.atadenoise"
plugin not available: "avfilter.bs2b"
plugin not available: "avfilter.bwdif"
plugin not available: "avfilter.deblock"
plugin not available: "avfilter.dedot"
plugin not available: "avfilter.deflate"
plugin not available: "avfilter.derain"
plugin not available: "avfilter.doubleweave"
plugin not available: "avfilter.field"
plugin not available: "avfilter.framestep"
plugin not available: "avfilter.fspp"
plugin not available: "avfilter.graphmonitor"
plugin not available: "avfilter.hqdn3d"
plugin not available: "avfilter.inflate"
plugin not available: "avfilter.lagfun"
plugin not available: "avfilter.loudnorm"
plugin not available: "avfilter.mcdeint"
plugin not available: "avfilter.random"
plugin not available: "avfilter.removegrain"
plugin not available: "avfilter.separatefields"
plugin not available: "avfilter.shuffleplanes"
plugin not available: "avfilter.sofalizer"
plugin not available: "avfilter.sr"
plugin not available: "avfilter.tmix"
plugin not available: "avfilter.w3fdif"
plugin not available: "avfilter.weave"
plugin not available: "avfilter.yadif"
plugin not available: "frei0r.baltan"
plugin not available: "frei0r.bgsubtract0r"
plugin not available: "frei0r.delay0r"
plugin not available: "frei0r.delaygrab"
plugin not available: "frei0r.facebl0r"
plugin not available: "frei0r.facedetect"
plugin not available: "frei0r.lightgraffiti"
plugin not available: "frei0r.lightgraffiti"
plugin not available: "frei0r.tehRoxx0r"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "ladspa"
plugin not available: "movit.unsharp_mask"
plugin not available: "timewarp"
plugin not available: "region"
=== /// CANNOT ACCESS SPEECH DICTIONARIES FOLDER
QQmlEngine::setContextForObject(): Object already has a QQmlContext
```

My `flatpak info org.kde.kdenlive` output is exactly identical to yours

**Comment by fabiscafe** [comment]
Possible Upstream Report: https://bugs.kde.org/show_bug.cgi?id=443574

Might crash for everyone with working hardware accelerated playback (and enabled setting)

**Comment by mattipulkkinen** [comment]
I tested that now by removing the Kdenlive flatpak, and then the whole ~/.var/app/org.kde.kdenlive directory. Then I reinstalled the Kdenlive flatpak, and it started up just fine. After enabling that GPU processing option in the playback settings, Kdenlive started crashing again. I don't remember enabling that before, but at least now I can get Kdenlive working again.

So, in summary, the bug seems to be upstream after all. I'll close this report because there's already an [upstream report](https://bugs.kde.org/show_bug.cgi?id=443574).
